nomos-services/network/src/backends/libp2p.rs

```diff
@@ -97,6 +95,18 @@ impl NetworkBackend for Libp2p {
         };
         overwatch_handle.runtime().spawn(async move {
             use tokio_stream::StreamExt;
+            let mut stream = if let Ok(Some(stream)) = mixnet_client.run().await {
+                Box::pin(stream.filter_map(|r| match r {
+                    Ok(msg) => Some(msg),
+                    Err(e) => {
+                        tracing::error!("error while receiving from mixnet: {e:?}");
+                        None
+                    }
+                })) as Pin<Box<dyn Stream<Item=Vec<u8>> + Send>>
+            } else {
+                tracing::error!("Could not quickstart mixnet stream");
+                Box::pin(stream::empty())
```

Comment:
Not sure if this need to be just an error or we should actually panic. If the service rely on it 🤔 . At least in the case where we actually return an error:

```rust
Ok(Some(stream)) => stream,
Ok(None) => log_error,
Err(e) => panic!
```

What do you think?

Reply:
Yes, returning an empty stream from the client seems like a cleaner API option.

Reply:
I agree. I'm working on this.

Reply:
6a5e2e9 It took some time to struggle with the compiler :)

I adopted `thiserror` to avoid the following error when returning either `stream::empty()` or `stream::unfold` from mixclient, if the stream type is `Stream<Result<_, Box<dyn Error>>>`.
```
error[E0277]: `(dyn StdError + 'static)` cannot be sent between threads safely
  --> mixnet/client/src/config.rs:29:32
   |
29 |             Self::Sender => Ok(stream::empty().boxed()),
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn StdError + 'static)` cannot be sent between threads safely
   |
```

Reply:
You just need to add `dyn Error + Send` in the code. I think it is better to use thiserror. But I wouldn't like the PR to keep growing too much. We can do that as a refactor later on.